app/cdap/components/GridTable/components/TransformationToolbar/index.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+ *  Copyright © 2022 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+import { IconButton, Typography, Tooltip, TextField } from '@material-ui/core';
+import Box from '@material-ui/core/Box';
+import { default as React, useState } from 'react';
+import NestedMenu from 'components/GridTable/components/NestedMenu';
+import { ITransformationToolBarProps } from 'components/GridTable/components/TransformationToolbar/types';
+import {
+  Divider,
+  Expand,
+  LongDivider,
+} from 'components/GridTable/components/TransformationToolbar/iconStore';
+import { useStyles } from 'components/GridTable/components/TransformationToolbar/styles';
+import FunctionToggle from 'components/GridTable/components/FunctionNameToggle';
+import { nestedMenuOptions } from 'components/GridTable/components/TransformationToolbar/utils';
+import { IMenuItem } from 'components/GridTable/components/MenuItemComponent/types';
+
+export default function({
+  columnType,
+  submitMenuOption,
+  setShowBreadCrumb,
+  showBreadCrumb,
+}: ITransformationToolBarProps) {
+  const classes = useStyles();
+  const [showName, setShowName] = useState<boolean>(false);
+  const [anchorElement, setAnchorElement] = useState<HTMLElement[]>(null);
+  const [selectedMenuOptions, setSelectedMenuOptions] = useState<IMenuItem[]>([]);
+
+  const toggleMenu = () => {
+    setSelectedMenuOptions([]);
+    setAnchorElement(null);
+  };
+
+  return (
+    <Box className={classes.iconContainer} data-testid="transformations-toolbar-container">
+      <Box className={classes.container} data-testid="nested-menu-container">
+        {nestedMenuOptions?.map((eachOption, optionIndex) => {
+          return (
+            <>
+              <Box
+                className={classes.functionNameWrapper}
+                data-testid={`toolbar-icon-${eachOption.title
+                  .toLowerCase()
+                  .split(' ')
+                  .join('-')}`}
+              >
+                {eachOption.options?.length ? (
```

Comment:
1. Don't use a ternary with such long code blocks - it's too easy to lose track. You could move the rendering to a function: `condition ? renderA() : renderB()`
2. Most of the code between the 2 options looks like same - could it be consolidated with only the differences handled conditionally?

Reply:
Done